README
Only tr13240 is python3 friendly - if you copy scirpts, do so from there
it also uses PILlow (Python Image Library) which I had installed as
replacement for lib_py_jpeg which is retired because it does not work
with python3



HISTORY
TAG: gxdload-6-0-18-1
DATE: 02/21/2022
STAFF: lec
CHANGES:
wts2-767/mgi_notechunk/mgi_note merge

TAG: gxdload-6-0-16-10
TAG: gxdload-6-0-16-9
TAG: gxdload-6-0-16-8
TAG: gxdload-6-0-16-7
TAG: gxdload-6-0-16-6
TAG: gxdload-6-0-16-5
TAG: gxdload-6-0-16-4
DATE: 04/15/2021
STAFF: lec
CHANGES:
TR13240/update IMPC lacZ data in GXD/J:228563
processing assays
delete obsolete Image records

TAG: gxdload-6-0-16-3
TAG: gxdload-6-0-16-2
TAG: gxdload-6-0-16-1
DATE: 03/25/2021
STAFF: sc
CHANGES: added tr13240 directory to load lacz images

TAG: gxdload-6-0-12-3
TAG: gxdload-6-0-12-2
TAG: gxdload-6-0-12-1
DATE: 07/13/2018, 07/23/2018
STAFF: lec
CHANGES:
1) TR12900/Allen developing brain paper

TAG: gxdload-6-0-11-1
DATE: 11/01/2017
STAFF: lec
CHANGES:
1) TR12649/GUDMAP3

TAG: gxdload-6-0-10-4
DATE: 10/20/2017
STAFF: sc
CHANGES: TR12250 littriage project
1) create insitu template script and add call to BIB_updateWFStatusGXD
    use this script as starting point for future GXD adhoc loads

TAG: gxdload-6-0-10-3
TAG: gxdload-6-0-10-2
TAG: gxdload-6-0-10-1
DATE: 09/14/2017
STAFF: lec
CHANGES:
1) TR12649/GUDMAP3

TAG: gxdload-6-0-9-6
TAG: gxdload-6-0-9-5
TAG: gxdload-6-0-9-4
TAG: gxdload-6-0-9-3
TAG: gxdload-6-0-9-2
DATE: 04/27/2017, 05/02/2017
STAFF: lec
CHANGES:
TR12504/IMPC/lacz update
tr12504/tr12504genotype.sh

TAG: gxdload-6-0-9-1
DATE: 04/06/2017
STAFF: lec
CHANGES:
TR12504/IMPC/lacz update
tr12504/tr12504genotype.sh

TAG: gxdload-6-0-8-5
TAG: gxdload-6-0-8-4
TAG: gxdload-6-0-8-3
TAG: gxdload-6-0-8-2
DATE: 03/27/2017
STAFF: lec
CHANGES:
TR12504/IMPC/lacz update
images loaded into MGI/GXD/EI/Image module (image stubs)

TAG: gxdload-6-0-8-1
DATE: 03/28/2017
STAFF: lec
CHANGES:
TR12504/IMPC/lacz update
images loaded into pixeldb

TAG: gxdload-6-0-7-8
TAG: gxdload-6-0-7-7
TAG: gxdload-6-0-7-6
TAG: gxdload-6-0-7-5
TAG: gxdload-6-0-7-4
TAG: gxdload-6-0-7-3
TAG: gxdload-6-0-7-2
TAG: gxdload-6-0-7-1
DATE: 12/29/2016, 02/07/2017
STAFF: lec
CHANGES:
TR12491/GXD/Lewandowski et al

TAG: gxdload-6-0-4-1
DATE: 04/25/2016, 05/05/2016
STAFF: lec
CHANGES:
1) TR12026/lacz
2) removed old TR directories
3) tr12026/tr12026images.sh copied from tr11471 but not used

TAG: gxdload-5-1-7-6
DATE: 03/07/2014
STAFF: lec
CHANGES:
1) TR11471/GUDMAP2/immunohistochemistry

TAG: gxdload-5-1-7-5
TAG: gxdload-5-1-7-4
TAG: gxdload-5-1-7-3
TAG: gxdload-5-1-7-2
TAG: gxdload-5-1-7-1
DATE: 01/28/2014, 01/30/2014, 02/06/2014
STAFF: lec
CHANGES:
1) TR11471/GUDMAP2/images only

TAG: gxdload-5-0-0-14
TAG: gxdload-5-0-0-13
TAG: gxdload-5-0-0-12
TAG: gxdload-5-0-0-11
TAG: gxdload-5-0-0-10
TAG: gxdload-5-0-0-9
TAG: gxdload-5-0-0-8
TAG: gxdload-5-0-0-7
TAG: gxdload-5-0-0-6
TAG: gxdload-5-0-0-5
TAG: gxdload-5-0-0-4
TAG: gxdload-5-0-0-3
TAG: gxdload-5-0-0-2
TAG: gxdload-5-0-0-1
DATE: 05/15/2012
DATE: 05/14/2012
DATE: 05/10/2012
DATE: 05/01/2012
DATE: 04/25/2012
DATE: 04/05/2012
STAFF: lec
CHANGES:
1) TR10976/images

TAG: gxdload-4-4-2-9
TAG: gxdload-4-4-2-8
TAG: gxdload-4-4-2-7
DATE: 03/15/2012
DATE: 03/14/2012
DATE: 02/27/2012
STAFF: lec
CHANGES:
1) tr10976/genepaint
2) 10900/10976/replacenotes
3) 10900/10976/removealias

TAG: gxdload-4-4-2-6
TAG: gxdload-4-4-2-5
TAG: gxdload-4-4-2-4
DATE: 12/15/2011, 12/19/2011
STAFF: lec
CHANGES:
1) tr10923/probes

TAG: gxdload-4-4-2-3
TAG: gxdload-4-4-2-2
TAG: gxdload-4-4-2-1
DATE: 10//11/2011
STAFF: lec
CHANGES:
1) tr10872/Tamplin/J:143778

TAG: gxdload-4-4-1-12
TAG: gxdload-4-4-1-11
TAG: gxdload-4-4-1-10
TAG: gxdload-4-4-1-9
DATE: 10/03/2011, 10/05/2011, 10/06/2011
STAFF: lec
CHANGES:
1) tr10680/J:168654
2) tr10840/Surani/J:174767

TAG: gxdload-4-4-1-8
TAG: gxdload-4-4-1-7
TAG: gxdload-4-4-1-6
TAG: gxdload-4-4-1-5
DATE: 07/13/2011
STAFF: lec
CHANGES:
1) tr10629

TAG: gxdload-4-4-1-4
DATE: 04/06/2011
STAFF: dbm
CHANGES:
1) Added tr10537assayload.sh to get the assays loaded.
2) Updated tr10537.config with new settings for the assay load

TAG: gxdload-4-4-1-3
DATE: 02/22/2011
STAFF: dbm
CHANGES:
1) Added tr10537.config and tr10537imageload.sh to get the images loaded.

TAG: gxdload-4-4-1-2
DATE: 12/29/2010
STAFF: dbm
CHANGES:
1) Added tr10449.config with new settings for the assay load.
2) Added tr10449assayload.sh to get the assays loaded.

TAG: gxdload-4-4-1-1
DATE: 12/21/2010
STAFF: dbm
CHANGES:
1) Added tr10407.config with new settings for the assay load.
2) Added tr10407assayload.sh to get the assays loaded.

TAG: gxdload-4-4-0-1
DATE: 09/08/2010
STAFF: dbm
CHANGES:
1) Added tr9695assayload.sh to get the assays loaded.
2) Updated tr9695.config with new settings for the assay load

TAG: gxdload-4-3-5-1
DATE: 08/02/2010
STAFF: dbm
CHANGES:
1) Added tr9695.config and tr9695imageload.sh to get the images loaded.

TAG: gxdload-4-3-4-4
TAG: gxdload-4-3-4-3
TAG: gxdload-4-3-4-2
TAG: gxdload-4-3-4-1
DATE: 05/11/2010
STAFF: lec
CHANGES: add TR10161/J:157819
         add TR10159/J:157352

TAG: gxdload-4-2-0-5
TAG: gxdload-4-2-0-4
TAG: gxdload-4-2-0-3
DATE: 01/21/2009
DATE: 01/15/2009
STAFF: lec
CHANGES: add TR9458/J:141291 GXD load

TAG: gxdload-4-2-0-2
TAG: gxdload-4-2-0-1
DATE: 01/13/2009
DATE: 01/07/2009
STAFF: lec
CHANGES: add TR9417/J:139177 GXD load

TAG: gxdload-3-5-4-1
DATE: 10/03/2007
STAFF: dbm
CHANGES: New product in CVS repository



tr12900/insitu.py
#!/usr/local/bin/python

#
# Program: insitu.py
#
# Original Author: Lori Corbani
#
# Purpose:
#
#	To translate tr12900 LoadFile.txt file
#
# Requirements Satisfied by This Program:
#
# Usage:
#
#	insitu.py
#
# Envvars:
#
# Inputs:
#
#	a tab-delimited file in the format:
#		field 1: Probe ID                            : MGI:xxx
#		field 2: ProbePrepType                       : RNA
#		field 3: Hybridization                       : Antisense
#		field 4: LabelledWith                        : digoxigenin
#		field 5: VisualizedWith                      : Alkaline phosphatase
#		field 6: MGI Marker ID                       : MGI:xxx
#		field 7: Reference                           : J:215487
#		field 8: AssayType                           : RNA in situ
#		field 9: ReporterGene                        : null
#		field 10: AssayNote                          : 
#		field 11: CreatedBy                          : cms
#		field 12,21,30,39,48,57,66: SpecimenLabel    :
#		field 13,22,31,40,49,58,67: Genotype         : MGI:2166311
#		field 14,23,32,41,50,59,68: Age              :
#		field 15,24,33,42,51,60,69: AgeNote          : null
#		field 16,25,34,43,52,61,70: Sex              : Not Specified for youngest 2 ages; Male for rest
#		field 17,26,35,44,53,62,71: Fixation         : Fresh Frozen
#		field 18,27,36,45,54,63,72: Embedding Method : Cryosection
#		field 19,28,37,46,55,64,73: Hybridization    : section
#		field 20,29,38,47,56,65,74: SpecimenNote     : null
#		field 75,81,93, 99,105,111: Strength      : Present (n=10,781), Absent (n=2,645)
#		field 76,82,94,100,106,112: Pattern      : Not Specified (n=10,781), Not Applicable (n=2,645)
#		field 77,83,95,101,107,113: Structure    : EMAPA:16894
#		field 78,84,96,102,108,114: TS           : 
#		field 79,85,97,103,109,115: Note         : null
#		field 80,86,98,104,110,116: Image        : null
#
# Outputs:
#
#       4 tab-delimited files:
#
#	In_Situ_probeprep.txt
#	In_Situ_assay.txt
#	In_Situ_specimen.txt
#	In_Situ_results.txt
#	
#       Error file
#
# Exit Codes:
#
# Assumes:
#
# Bugs:
#
# Implementation:
#

import sys
import os
import string
import db

TAB = '\t'		# tab
CRT = '\n'		# carriage return/newline
NULL = ''

inSpecimenFile = ''	# file descriptor

prepFile = ''           # file descriptor
assayFile = ''          # file descriptor
specimenFile = ''       # file descriptor
resultsFile = ''        # file descriptor

datadir = os.environ['PROJECTDIR']

inSpecimenFileName = os.environ['INPUT_FILE']

prepFileName = datadir + '/In_Situ_probeprep.txt'
assayFileName = datadir + '/In_Situ_assay.txt'
specimenFileName = datadir + '/In_Situ_specimen.txt'
resultsFileName = datadir + '/In_Situ_results.txt'

# Purpose: prints error message and exits
# Returns: nothing
# Assumes: nothing
# Effects: exits with exit status
# Throws: nothing

def exit(
    status,          # numeric exit status (integer)
    message = None   # exit message (string)
    ):

    if message is not None:
        sys.stderr.write('\n' + str(message) + '\n')
 
    sys.exit(status)
 
# Purpose: initialize
# Returns: nothing
# Assumes: nothing
# Effects: initializes global variables
#          exits if files cannot be opened
# Throws: nothing

def init():
    global inSpecimenFile
    global prepFile, assayFile, specimenFile, resultsFile
 
    try:
        inSpecimenFile = open(inSpecimenFileName, 'r')
    except:
        exit(1, 'Could not open file %s\n' % inSpecimenFileName)

    try:
        prepFile = open(prepFileName, 'w')
    except:
        exit(1, 'Could not open file %s\n' % prepFileName)

    try:
        assayFile = open(assayFileName, 'w')
    except:
        exit(1, 'Could not open file %s\n' % assayFileName)

    try:
        specimenFile = open(specimenFileName, 'w')
    except:
        exit(1, 'Could not open file %s\n' % specimenFileName)

    try:
        resultsFile = open(resultsFileName, 'w')
    except:
        exit(1, 'Could not open file %s\n' % resultsFileName)

    return

# Purpose:  processes data
# Returns:  nothing
# Assumes:  nothing
# Effects:  writes data to output files
# Throws:   nothing

def process():

    assayKey = 1
    specimenLookup = {}
    specimenProbeLookup = {}

#		field 1: Probe ID                            : MGI:xxx
#		field 2: ProbePrepType                       : RNA
#		field 3: Hybridization                       : Antisense
#		field 4: LabelledWith                        : digoxigenin
#		field 5: VisualizedWith                      : Alkaline phosphatase
#		field 6: MGI Marker ID                       : MGI:xxx
#		field 7: Reference                           : J:215487
#		field 8: AssayType                           : RNA in situ
#		field 9: ReporterGene                        : null
#		field 10: AssayNote                          : 
#		field 11: CreatedBy                          : cms

    for line in inSpecimenFile.readlines():

        tokens = string.split(line[:-1], TAB)
	probeID = tokens[0]
	markerID = tokens[5]

	prepFile.write(str(assayKey) + TAB + \
		probeID + TAB + \
		'RNA' + TAB + \
		'Antisense' + TAB + \
		'digoxigenin' + TAB + \
		'Alkaline phosphatase' + CRT)

	# write the assay information

        assayFile.write(str(assayKey) + TAB + \
            markerID + TAB + \
            'J:215487' + TAB + \
            'RNA in situ' + TAB + \
	    TAB + \
	    tokens[9] + TAB + \
            'cms' + CRT)

#		field 12,21,30,39,48,57,66: SpecimenLabel    :
#		field 13,22,31,40,49,58,67: Genotype         : MGI:2166311
#		field 14,23,32,41,50,59,68: Age              :
#		field 15,24,33,42,51,60,69: AgeNote          : null
#		field 16,25,34,43,52,61,70: Sex              : Not Specified for youngest 2 ages; Male for rest
#		field 17,26,35,44,53,62,71: Fixation         : Fresh Frozen
#		field 18,27,36,45,54,63,72: Embedding Method : Cryosection
#		field 19,28,37,46,55,64,73: Hybridization    : section
#		field 20,29,38,47,56,65,74: SpecimenNote     : null

        specimenKey = 1
	for s in (12,21,30,39,48,57,66):
	    s = s - 1

	    specimenLabel = tokens[s]
	    age = tokens[s + 2]

	    if age in ('embryonic day 11.5', 'embryonic day 13.5'):
	        sex = 'Not Specified'
            else:
	        sex = 'Male'

	    specimenFile.write(str(assayKey) + TAB + \
	        str(specimenKey) + TAB + \
	        specimenLabel + TAB + \
	        'MGI:2166311' + TAB + \
	        age + TAB + \
	        TAB + \
	        sex + TAB + \
	        'Fresh Frozen' + TAB + \
	        'Cryosection' + TAB + \
	        'section' + TAB + \
	        CRT)

#		field 75,81,87,93, 99,105,111: Strength      : Present (n=10,781), Absent (n=2,645)
#		field 76,82,88,94,100,106,112: Pattern      : Not Specified (n=10,781), Not Applicable (n=2,645)
#		field 77,83,89,95,101,107,113: Structure    : EMAPA:16894
#		field 78,84,90,96,102,108,114: TS           : 
#		field 79,85,91,97,103,109,115: Note         : null
#		field 80,86,92,98,104,110,116: Image        : null

	    if s == 11:
	        r = 75
	    elif s == 20:
	        r = 81
	    elif s == 29:
	        r = 87
	    elif s == 38:
	        r = 93
	    elif s == 47:
	        r = 99
	    elif s == 56:
	        r = 105
	    elif s == 65:
	        r = 111

            resultKey = 1
	    r = r - 1
	    resultsFile.write(str(assayKey) + TAB + \
	        str(specimenKey) + TAB + \
	        str(resultKey) + TAB + \
	        tokens[r] + TAB + \
	        tokens[r + 1] + TAB + \
	        'EMAPA:16894' + TAB + \
	        tokens[r + 3] + TAB + \
	        TAB + \
	        CRT)
    
            specimenKey = specimenKey + 1

	assayKey = assayKey + 1

    inSpecimenFile.close()

#
# Main
#

init()
process()
exit(0)

